core/src/main/java/org/elasticsearch/action/main/MainResponse.java

```diff
@@ -114,4 +117,24 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.endObject();
         return builder;
     }
+
+    private final static ObjectParser<MainResponse, Void> PARSER = new ObjectParser<>(MainResponse.class.getName(),
+            () -> new MainResponse());
+
+    static {
+        PARSER.declareString((response, value) -> response.nodeName = value, new ParseField("name"));
+        PARSER.declareString((response, value) -> response.clusterName = new ClusterName(value), new ParseField("cluster_name"));
+        PARSER.declareString((response, value) -> response.clusterUuid = value, new ParseField("cluster_uuid"));
+        PARSER.declareString((response, value) -> {}, new ParseField("tagline"));
+        PARSER.declareObject((response, value) -> {
+            response.build = new Build((String) value.get("build_hash"), (String) value.get("build_date"),
+                    (boolean) value.get("build_snapshot"));
+            response.version = Version.fromString((String) value.get("number"));
+            response.available = true;
+        }, (parser, context) -> parser.map(), new ParseField("version"));
```

Comment:
No, ObjectParser has an `ignoreUnknownFields` flag that, if set, skips any undeclared field. I think that if we set that for response parsers we are good with adding new fields there. If we need this to be globally switchable (e.g. from a setting) I think we can add that to Object parser.